hydra-main/src/main/java/com/addthis/hydra/query/web/HttpQueryCallHandler.java

```diff
@@ -65,8 +63,10 @@ public static ChannelFuture handleQuery(ChannelHandler queryToQueryResultsEncode
         if ((dir != null) && !job.endsWith(dir)) {
             String[] jobs = job.split(",");
             String[] dirs = dir.split(",");
-            job = Arrays.stream(jobs).flatMap(subJob ->  Arrays.stream(dirs).map(subDir -> subJob + '/' + subDir))
+            try (Stream<String> stream = Arrays.stream(jobs)) {
+                stream.flatMap(subJob ->  Arrays.stream(dirs).map(subDir -> subJob + '/' + subDir))
                         .collect(Collectors.joining(","));
```

Comment:
Same deal as before. Strings to strings to strings. There's no reason to fret over "leaking" the stream any more than there is to to be concerned about `String[] jobs`.